wandb/sdk/internal/tpu.py

```diff
@@ -0,0 +1,70 @@
+from __future__ import absolute_import
+
+import logging
+import os
+from subprocess import PIPE, Popen
+import threading
+
+import wandb
+
+logger = logging.getLogger(__name__)
+
+
+class TPUProfiler(object):
+    def __init__(self):
+        try:
+            import cloud_tpu_profiler  # type: ignore
+
+            del cloud_tpu_profiler  # flake8
+            self._enabled = True
+        except ImportError:
+            wandb.termwarn(
+                "cloud_tpu_profiler is not installed. "
+                "TPU stats will not be captured."
+            )
+            logger.warn(
+                "cloud_tpu_profiler is not installed. "
+                "TPU stats will not be captured."
+            )
+            self._enabled = False
+            return
+        self._tpu_utilization = 0
+        self._stop_thread = False
+        self._thread = threading.Thread(target=self._thread_body, daemon=True)
+        self._thread.start()
+
+    def _thread_body(self):
+        while not self._stop_thread:
+            self._tpu_utilization = self._get_tpu_utilization()
+
+    def _get_tpu_utilization(self):
+        # blocking
+        args = [
+            "capture_tpu_profile",
+            "--tpu=" + os.environ["TPU_NAME"],
+            "--monitoring_level=2",
+            "--num_queries=1",
+            "--duration_ms=100",
+        ]
+        try:
+            p = Popen(args, stdout=PIPE, stderr=None, universal_newlines=True)
```

Comment:
Seems fishy there's no way to just call this directly in the `cloud_tpu_profiler` library.  Doing it this way could open us up to weird system path issues and the overhead of spawning a new process every time we capture metrics is less than ideal.

Reply:
Thats what I originally had, but weird pickling issues when the profiler is called from the backend process. Also tried keeping the process alive and and streaming the output from it, but not reliable.